state/protocol/badger/mutator.go

```diff
@@ -433,6 +459,10 @@ func (m *FollowerState) insert(ctx context.Context, candidate *flow.Block, last
 		return fmt.Errorf("could not process service events: %w", err)
 	}
 
+	qc := candidate.Header.QuorumCertificate()
+	_, err = m.qcs.ByBlockID(qc.BlockID)
+	qcAlreadyInserted := err == nil
```

Comment:
The attempted insertion will perform a read first anyway, so this is unnecessary IMO (we're just adding an extra read). 

Also, if we were to keep this we should check for `err == storage.ErrAlreadyExists` and propagate other errors.

Reply:
this was meant for emitting only one `BlockProcessable` event, but we could do it differently.

Reply:
https://github.com/onflow/flow-go/pull/3949/commits/a42afb0d3ad5d09d473b81730b0c13e41fd310b8